File: cpp/204. CountPrimes.cpp
class Solution {
public:
    int countPrimes(int n) {
        int count=0;
        bool prime[n+1];
        memset(prime,true,sizeof(prime));
        for(int p=2;p*p<n;p++)
        {
            if(prime[p]==true)
            {
                for(int i=p*p;i<n;i+=p)
                        prime[i]=false;
            }
        }
        for(int i=2;i<n;i++)
            if(prime[i]==true)
                count++;
        return count;
    }
};

File: py/204. Count Primes.py
def countPrimes(self, n: int) -> int:
        
        if n < 3:
            return 0
        
        
        # 1. boolean Table for prime number setting all number as prime = 1
        isPrime = [1] * (n)   
        isPrime[0] = isPrime[1] = 0
        count = 0              

        m = int(n**0.5)+1
        
        for i in range(2,m):
            if isPrime[i] == 1:
                for j in range(i+i,n,i):  # next multiple
                    isPrime[j] = 0           # non prime
        
        for i in isPrime:
            if i == 1:
                count += 1
                
        return count
